core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -0,0 +1,106 @@
+                             namespaces: Set[KubernetesNamespaceName],
+                             labels: LabelMap,
+                             nodepools: Set[Nodepool]
+                            )
+
+object KubernetesCluster {
+  implicit class EnrichedKubernetesCluster(cluster: KubernetesCluster) {
+    def getGKEClusterId: KubernetesClusterId = KubernetesClusterId(cluster.googleProject, cluster.location, cluster.clusterName)
+  }
+}
+
+case class SaveKubernetesCluster(googleProject: GoogleProject,
+                                 clusterName: KubernetesClusterName,
+                                 location: Location,
+                                 status: KubernetesClusterStatus,
+                                 serviceAccountInfo: WorkbenchEmail,
+                                 samResourceId: KubernetesClusterSamResource,
+                                 auditInfo: AuditInfo,
+                                 labels: LabelMap,
+                                 initialNodepool: Nodepool) //the clusterId specified here isn't used, and will be replaced by the id of cluster saved beforehand
+
+case class KubernetesClusterSamResource(resourceId: String)
+
+case class KubernetesClusterAsyncFields(apiServerIp: Option[KubernetesApiServerIp],
+                                        networkInfo: Option[NetworkFields]
+                                       )
+
+case class NetworkFields(networkName: NetworkName,
+                         subNetworkName: SubnetworkName,
+                         subNetworkIpRange: IpRange)
+
+
+//TODO: possibly consolidate Kubernetes status when monitoring is implemented
+sealed trait KubernetesClusterStatus extends EnumEntry with Product with Serializable
+object KubernetesClusterStatus extends Enum[KubernetesClusterStatus] {
+  val values = findValues
+
+  // NOTE: Remember to update the definition of this enum in Swagger when you add new ones
+  case object Status_Unspecified extends KubernetesClusterStatus
+  case object Provisioning extends KubernetesClusterStatus
+  case object Running extends KubernetesClusterStatus
+  case object Reconciling extends KubernetesClusterStatus
+  case object Stopping extends KubernetesClusterStatus
+  case object Error extends KubernetesClusterStatus
+  case object Degraded extends KubernetesClusterStatus
+}
+
+//TODO: possibly consolidate Kubernetes status when monitoring is implemented
+sealed trait NodepoolStatus extends EnumEntry with Product with Serializable
+object NodepoolStatus extends Enum[NodepoolStatus] {
+  val values = findValues
+
+  // NOTE: Remember to update the definition of this enum in Swagger when you add new ones
+  case object Status_Unspecified extends NodepoolStatus
+  case object Provisioning extends NodepoolStatus
+  case object Running extends NodepoolStatus
+  case object Running_with_Error extends NodepoolStatus //consistent with Degraded from KubernetesClusterStatus
```

Comment:
`RunningWithError` instead?

Reply:
I'd prefer in scala code, casing is right...and add a 
```
def asString: String
```
To `NodepoolStatus`, and define it in all product types...

Reply:
These words (and thus the strings the underlying enum implementation provides) map directly to the google container statuses.

Reply:
Why do our Scala code names have to mirror Google's style exactly?

Reply:
yea. I know this is how current code has been. But we really shouldn't...We should never rely on the `case object` name for its String representation. It's very error, if one day someone decides to rename the case object, they shouldn't need to worry that "oh, would this break the contract as well?"

Reply:
The premise behind the `Enum` implementation we are using is that your case object name maps via `.withName`, which makes decoding/encoding trivial. Is it really worth decoupling these, only to move the coupling to a mapping table we maintain? That would require hardcoded strings being returned from an `asString` coupled to a map. it seems like unecessary code to maintain instead of using built-in enum features. Let's jump on a call at some point

Reply:
Yes it's worth the effort. And we shouldn't use `withName` because it can throw exception if one day google decides to add a new status. An example of the approach I'm talking about https://github.com/DataBiosphere/leonardo/blob/87db66a47d2ba1e2946505984fbd5cedded7415a/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dataprocModels.scala#L96.